src/ensembl/src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields.tsx

```diff
@@ -1,115 +1,148 @@
 import React, { useState, useEffect } from 'react';
```

Comment:
Very confused by the name of this component now. It says "Checkbox with textfields", but now also contains logic about "files"

Reply:
I did notice it but I thought it would be better to refactor on a different PR than keep on extending this one.